CORE/pages/core_page_element_methods.rb

```diff
@@ -12,7 +12,7 @@ def group(*args)
         ElementGroup.new(args)
     end
     
-    def add_element(name, options, element_class)
+    def add_element(name, options, element_class = CoreElement)
```

Comment:
This and the subsequent uses of `add_element()` on pages are going over my head. Want to discuss this.

Reply:
The general gist of what's going on here is i'm defaulting add_element to create a CoreElement. This allows us to use somewhat nicer language when we're creating parent elements

In the copy of the registration page that uses the parents, i create the page sections as a generic element type since we only care that they're present.
```ruby
    account_creation_form = add_element(:account_creation_form, element_type: :form, id: 'account-creation_form')
    personal_information_section = add_element(:personal_information_section, element_type: :div, parent: account_creation_form, class: 'account_creation', index: 0)

```
This defines a generic element called `account_creation_form`  I immediately use it on the following line to constrain my subsequent search for the `personal_information_section` to the `account_creation_form` with  `parent: account_creation_form`, I also do the same with the address section and date of birth section.

Now I can use way simpler selectors in most places in the page like
```ruby
    add_static_text(:your_personal_information, element_type: :h3, parent: personal_information_section, class: 'page-subheading')
```
and
```ruby
    add_static_text(:password_hint, element_type: :span, parent: personal_information_section, class: 'form_info')

```
in place of
```ruby
    add_static_text(:your_personal_information, element_type: :h3, xpath: "//form[@id='account-creation_form']/div[1]/h3")

```
and
```ruby
    add_static_text(:password_hint, element_type: :span, xpath: "//div[@class='account_creation'][1]/div[5]/span")
```